internal/common/objects/BaseBiker.go

```diff
@@ -181,6 +197,40 @@ func (bb *BaseBiker) GetResourceAllocationParams() ResourceAllocationParams {
 	return bb.allocationParams
 }
 
+// default implementation returns the id of the nearest lootbox
+func (bb *BaseBiker) ProposeDirection() uuid.UUID {
+	return bb.NearestLoot()
+}
+
+func (bb *BaseBiker) ToggleOnBike() {
+	bb.onBike = !bb.onBike
+}
+
+func (bb *BaseBiker) GetBikeStatus() bool {
+	return bb.onBike
+}
+
+// an agent will have to rank the agents that are trying to join and that they will try to
+func (bb *BaseBiker) DecideJoining(pendingAgents []uuid.UUID) map[uuid.UUID]bool {
+	decision := make(map[uuid.UUID]bool)
+	for _, agent := range pendingAgents {
+		decision[agent] = true
+	}
+	return decision
+}
+
+// this function will contain the agent's strategy on deciding which direction to go to
+// the default implementation returns an equal distribution over all options
+// this will also be tried as returning a rank of options
+func (bb *BaseBiker) FinalDirectionVote(proposals []uuid.UUID) map[uuid.UUID]float64 {
+	votes := make(map[uuid.UUID]float64)
+	totOptions := len(proposals)
+	for _, proposal := range proposals {
+		votes[proposal] = 1.0 / float64(totOptions)
```

Comment:
You're recasting every iteration. Just cast once outside the loop or, since it doesn't seem that the cardinality is important (just the ordinality), set these all equal to 1 to avoid the semi costly float conversion